tiledb/common/pmr.h

```diff
@@ -187,77 +187,7 @@ class list : public pmr_list<Tp> {
-
-template <class Tp>
-class vector : public pmr_vector<Tp> {
- public:
-  // This class exists to ensure that all uses of it are provided with a
-  // valid std::pmr based allocator. This is so that as we switch from
-  // std::vector to using this class we don't forget to provide the allocator
-  // which is quite easy to do.
-  //
-  // If these constructors look confusing, just know that all we're doing is
-  // copying the current definitions from cppreference and then adjusting types
-  // to require the PMR based allocator.
-
-  // I have absolutely no idea if all of these aliases are required. The
-  // allocator_type is the important one. I've copied the others just in
-  // case since I do know that PMR aware containers at least require
-  // allocator_type.
-  using value_type = typename pmr_vector<Tp>::value_type;
-  using allocator_type = typename pmr_vector<Tp>::allocator_type;
-  using size_type = typename pmr_vector<Tp>::size_type;
-  using difference_type = typename pmr_vector<Tp>::difference_type;
-  using reference = typename pmr_vector<Tp>::reference;
-  using const_reference = typename pmr_vector<Tp>::const_reference;
-  using pointer = typename pmr_vector<Tp>::pointer;
-  using const_pointer = typename pmr_vector<Tp>::const_pointer;
-  using iterator = typename pmr_vector<Tp>::iterator;
-  using const_iterator = typename pmr_vector<Tp>::const_iterator;
-  using reverse_iterator = typename pmr_vector<Tp>::reverse_iterator;
-  using const_reverse_iterator =
-      typename pmr_vector<Tp>::const_reverse_iterator;
-
-  // Delete all default constructors because they don't require an allocator
-  constexpr vector() noexcept(noexcept(allocator_type())) = delete;
-  constexpr vector(const vector& other) = delete;
-  constexpr vector(vector&& other) noexcept = delete;
-
-  // Delete non-allocator aware copy and move assign.
-  constexpr vector& operator=(const vector& other) = delete;
-  constexpr vector& operator=(vector&& other) noexcept = delete;
-
-  constexpr explicit vector(const allocator_type& alloc) noexcept
-      : pmr_vector<Tp>(alloc) {
-  }
-
-  constexpr vector(
-      size_type count, const Tp& value, const allocator_type& alloc)
-      : pmr_vector<Tp>(count, value, alloc) {
-  }
-
-  constexpr explicit vector(size_type count, const allocator_type& alloc)
-      : pmr_vector<Tp>(count, alloc) {
-  }
-
-  template <class InputIt>
-  constexpr vector(InputIt first, InputIt last, const allocator_type& alloc)
-      : pmr_vector<Tp>(first, last, alloc) {
-  }
-
-  constexpr vector(const vector& other, const allocator_type& alloc)
-      : pmr_vector<Tp>(other, alloc) {
-  }
-
-  constexpr vector(vector&& other, const allocator_type& alloc)
-      : pmr_vector<Tp>(other, alloc) {
-  }
-
-  constexpr vector(std::initializer_list<Tp> init, const allocator_type& alloc)
-      : pmr_vector<Tp>(init, alloc) {
-  }
-};
+using vector = std::vector<Tp, polymorphic_allocator<Tp>>;
```

Comment:
Yes that is the change I think I want to make... Let me ponder on it for a little while longer. This is blocked on me now.